lib/db_connection/connection.ex

```diff
@@ -65,6 +65,8 @@ defmodule DBConnection.Connection do
   @doc false
   @impl :gen_statem
   def init({mod, opts, pool, tag}) do
+    Process.flag(:trap_exit, true)
```

Comment:
Let's not trap exits by default and leave the connection to opt-in to that. Let's also add some documentation to the disconnect callback saying it will be called on terminate if trapping exits, WDYT?

Reply:
That makes sense. I made the option name a bit more generic because I noticed in the tests that `terminate/3` would also be called if i'm not trapping exits in some situations.